src/app/pages/courses-page/courses-page.component.ts

```diff
@@ -44,6 +50,16 @@ export class CoursesPageComponent implements OnInit {
   }
 
   private getCourses() {
-    this.courses$ = this.coursesService.getCourses$();
+    this.loading = true;
+
+    this.coursesService.getCourses$()
+      .pipe(finalize(() => this.loading = false))
+      .subscribe(
+        courses => {
+          this.courses = courses;
+          this.coursesForDisplay = courses;
+        },
+        () => alert('An error has occured during the courses loading'),
+      );
```

Comment:
As I can see you already start using services.

It will be nice to move loader logic to his own service. This will provide your app with ability to call/hide loader not only for courses update but although for login/logout actions e.t.c.

That's only advice, out of this module demand :)

Reply:
Yes, good point for implementation of  global app-level loader